### DISCUSSION TITLE
RTMP support

### DISCUSSION THREAD

**Discussion by kelvinaliyanto** [TheWidlarzGroup/react-native-video#4939] [Ideas]
### What problem are you trying to solve?

The user wants to play RTMP streaming content with react-native-video.

### Proposed solution

Add support for RTMP streaming.

### Additional context

Original message: "Hi, great work! But does it support RTMP streaming?"

**Comment by sabbyt** [comment]
@kelvinaliyanto This is built on top of react-native-video and supports rtmp and rtsp https://www.npmjs.com/package/react-native-vlc

**Comment by kelvinaliyanto** [comment]
@sabbyt I believe react-native-vlc only supports Android. Do you have any idea of other work supporting iOS? Thanks

**Comment by sabbyt** [comment]
@kelvinaliyanto haven't used it myself but this seems to be the iOS version https://github.com/xiongchuan86/react-native-vlcplayer

**Comment by imagine10255** [comment]
@sabbyt Do you have successful use? I am also looking for support rtmp suite

**Comment by asukiaaa** [comment]
I succeeded in viewing rtmp live stream on Android with using this lib.
https://github.com/NodeMedia/react-native-nodemediaclient

**Comment by edison-fe-html** [comment]
+1

**Comment by aeliusventure** [comment]
@asukiaaa - Is there any way to capture screenshot from live streaming?

**Comment by lewatt23** [comment]
@aeliusventure found any solution ?

**Comment by chaunhatquang** [comment]
> I succeeded in viewing rtmp live stream on Android with using this lib. https://github.com/NodeMedia/react-native-nodemediaclient

until now, I can't play rtmp althought status notification NetStream.Buffer.Full. Plz help me?

**Comment by endrits079** [comment]
Why is this closed, it is still wanted?

**Comment by deflexable** [comment]
any progress on this?

**Comment by bachngocnhat** [comment]
+1

**Comment by freeboub** [comment]
If someone wants to implement the feature, the expected changes are very similar to following PR: https://github.com/react-native-video/react-native-video/pull/2824
And an additionnal article for exoplayer integration: https://karthikponnam.medium.com/rtmp-player-android-using-exo-media-player-ac7a012b7a25

**Comment by andreyukD** [comment]
+1

**Comment by freeboub** [comment]
Need to add rtmp datasource support with plugin: media3-datasource-rtmp
see a very similar PR here: https://github.com/react-native-video/react-native-video/pull/3677
I tagged the feature as a good first issue. Another mandatory requierement is to have sample stream ...

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
